src/main/haskell/kore/src/Kore/Step/ExpandedPattern.hs

```diff
@@ -49,58 +59,72 @@ import           Kore.Predicate.Predicate
                  makeAndPredicate, makeFalsePredicate, makeTruePredicate,
                  substitutionToPredicate, unwrapPredicate )
 import qualified Kore.Predicate.Predicate as Predicate
-import           Kore.Step.PredicateSubstitution
-                 ( CommonPredicateSubstitution, PredicateSubstitution (..) )
 import           Kore.Unification.Data
 import           Kore.Variables.Free
                  ( pureAllVariables )
 
-{-|'ExpandedPattern' is a representation of a PureMLPattern that is easier
-to use when executing Kore. It consists of an "and" between a term, a
-predicate and a substitution
--}
+{- | @Predicated@ represents a value conditioned on a predicate.
 
-type ExpandedPattern level variable =
-    Predicated level variable (PureMLPattern level variable)
+@Predicated level variable child@ represents a @child@ conditioned on a
+@predicate@ and @substitution@ (which is a specialized form of predicate).
+
+The 'Applicative' instance conjoins the predicates and substitutions so that the
+result is conditioned on the combined predicates of the inputs. The combined
+predicate and substitution are /not/ normalized.
```

Comment:
IMO it's very easy to use the applicative stuff and to forget to normalize at the end, and I'm not confident that using non-normalized substitutions would always work out properly (though it might since substitutions are equalities). It would be nice to have a type-safe way of combining this stuff.